gum/arch-arm64/gumarm64relocator.c

```diff
@@ -346,58 +346,131 @@ gum_arm64_relocator_can_relocate (gpointer address,
     csh capstone;
     cs_err err;
     cs_insn * insn;
-    size_t count, i;
-    gboolean eoi;
+    GHashTable * targets_to_check, * checked_targets;
+    guint8 * code_cur, * address_cur;
+    guint code_size;
+    GHashTableIter iter;
+    gpointer key, value;
 
     err = cs_open (CS_ARCH_ARM64, CS_MODE_LITTLE_ENDIAN, &capstone);
     g_assert_cmpint (err, == , CS_ERR_OK);
     err = cs_option (capstone, CS_OPT_DETAIL, CS_OPT_ON);
     g_assert_cmpint (err, ==, CS_ERR_OK);
 
-    count = cs_disasm (capstone, rl.input_cur, 1024, rl.input_pc, 0, &insn);
-    g_assert (insn != NULL);
+    insn = cs_malloc(capstone);
+    targets_to_check = g_hash_table_new (NULL, NULL);
+    checked_targets = g_hash_table_new (NULL, NULL);
 
-    eoi = FALSE;
-    for (i = 0; i != count && !eoi; i++)
+    code_cur = rl.input_cur;
+    address_cur = rl.input_pc;
+    code_size = 1024;
+    gboolean tragic_exit = false;
+    while (true)
     {
-      cs_arm64 * d = &insn[i].detail->arm64;
+      gboolean go_ahead = true;
+      g_hash_table_add (checked_targets, code_cur);
```

Comment:
This feels like it could be more readable if allowed to breathe a bit, by having a blank line on each side.